mono/metadata/mono-perfcounters.c

```diff
@@ -415,60 +416,6 @@ predef_cleanup (ImplVtable *vtable)
 	perfctr_unlock ();
 }
 
-static guint64
-mono_determine_physical_ram_size (void)
-{
-#if defined (TARGET_WIN32)
-	MEMORYSTATUSEX memstat;
-
-	memstat.dwLength = sizeof (memstat);
-	GlobalMemoryStatusEx (&memstat);
-	return (guint64)memstat.ullTotalPhys;
-#elif defined (__NetBSD__) || defined (__APPLE__)
-#ifdef __NetBSD__
-	unsigned long value;
-#else
-	guint64 value;
-#endif
-	int mib[2] = {
-		CTL_HW,
-#ifdef __NetBSD__
-		HW_PHYSMEM64
-#else
-		HW_MEMSIZE
-#endif
-	};
-	size_t size_sys = sizeof (value);
-
-	sysctl (mib, 2, &value, &size_sys, NULL, 0);
-	if (value == 0)
-		return 134217728;
-
-	return (guint64)value;
-#elif defined (HAVE_SYSCONF)
-	guint64 page_size = 0, num_pages = 0;
-
-	/* sysconf works on most *NIX operating systems, if your system doesn't have it or if it
-	 * reports invalid values, please add your OS specific code below. */
-#ifdef _SC_PAGESIZE
-	page_size = (guint64)sysconf (_SC_PAGESIZE);
-#endif
-
-#ifdef _SC_PHYS_PAGES
-	num_pages = (guint64)sysconf (_SC_PHYS_PAGES);
-#endif
-
-	if (!page_size || !num_pages) {
-		g_warning ("Your operating system's sysconf (3) function doesn't correctly report physical memory size!");
-		return 134217728;
-	}
-
-	return page_size * num_pages;
-#else
-	return 134217728;
```

Comment:
Could we do better here?

Reply:
I'm not sure how to do better since this is case where the OS basically doesn't tell us, or we don't know the appropriate API.